gqcp/include/FockSpace/WaveFunction/WaveFunctionSelection.hpp

```diff
@@ -0,0 +1,76 @@
+// This file is part of GQCG-gqcp.
+// 
+// Copyright (C) 2017-2019  the GQCG developers
+// 
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#pragma once
+
+
+#include "Basis/TransformationMatrix.hpp"
+#include "FockSpace/SelectedFockSpace.hpp"
+#include "FockSpace/WaveFunction/WaveFunction.hpp"
+#include "Mathematical/Representation/Matrix.hpp"
+
+
+
+namespace GQCP {
+
+
+/**
+ *  A class that represents a selection of a wave function
+ */
+class WaveFunctionSelection {
+private:
+    std::vector<Configuration> configurations;
+    VectorX<double> coefficients;  // the coefficients for the ONV configurations
+
+public:
+    // CONSTRUCTORS
+    WaveFunctionSelection() = default;
+
+    /**
+     *  Construct a selected wave function
+     *
+     *  @param fock_space           the selected Fock space in which the wave function 'lives'
+     *  @param coefficients         the expansion coefficients
+     */
+    WaveFunctionSelection(const SelectedFockSpace& fock_space, const VectorX<double>& coefficients);
```

Comment:
It feels weird what this constructor is trying to do. Does it extract all the coefficients from the given `SelectedFockSpace`?

Instead, could you implement a designated constructor that sets the members, and then clarify the documentation for this one and implement this delegated one?